src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -2581,6 +2581,74 @@ theorem signExtend_eq_append_of_le {w v : Nat} {x : BitVec w} (h : w ≤ v) :
   cases hx : x.msb <;>
     simp [getElem_cast, hx, getElem_append, getElem_signExtend]
 
+/--
+The 'master theorem' for extracting bits from `(xhi ++  xlo)`,
+which performs a case analysis on the start index, length, and the lengths of `xlo, xhi`.
+· If the start index is entirely out of the `xlo` bitvector, then grab the bits from `xhi`.
+· If the start index is entirely contained in the `xlo` bitvector, then grab the bits from `xlo`.
+· If the start index is split between the two bitvectors,
+  then append `(w - start)` bits from `xlo` with `(len - (w - start))` bits from xhi.
+  Diagramatically:
```

Comment:
```suggestion
· If the start index is split between the two bitvectors, then append `(w - start)` bits from `xlo` with
  `(len - (w - start))` bits from xhi.
  Diagramatically:
```

Reply:
I disagree with this reflow, since it breaks the sentence at an unnatural location.